### PR TITLE
Always create a duplicated context in OpenTelemetry when executing client requests

### LINKED ISSUE #25336: OpenTelemetry Vertx issue - divided traces
### Describe the bug

When using nested/parallel vertx calls with reactive quarkus, issue with OpenTelemetry occurs. Traces are divided instead of being in single trace and some http post spans are missing.

For parallel calls all of them are missing (if using parallel calls greater than 2)
For cascaded calls the dependent http posts spans are missing.

### Expected behavior

One single trace with http post calls included.

### Actual behavior

**Cascaded calls (two parallel calls and on result called third):**
all traces:
bugged_all_spans

detached trace:
bugged_detached_trace

main trace:
bugged_main_trace

**3 Parallel calls:**
all traces:
image

first trace with http post
image

other 2 traces (both same, uploading only one image)
image

### How to Reproduce?

1. Have standard jaeger all in one with collector listening on 14250
2. Download and run project from an attachment (can be simply runned from idea via quarkus plugin)
[quarkus-vertx-otel-bug-001.zip](https://github.com/quarkusio/quarkus/files/8610712/quarkus-vertx-otel-bug-001.zip)
3. Call get on localhost:8080/parallel or localhost:8080/cascaded
4. Observe the traces in jaeger ui 

### Output of `uname -a` or `ver`

Linux optimus-insurance-get-5b9fbc85c4-hdhhk 5.4.181-99.354.amzn2.x86_64 #1 SMP Wed Mar 2 18:50:46 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

graalvm-ce-java11-linux-amd64-20.3.0

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Runnable from intellij idea quarkus plugin simply

### Additional information

_No response_

### LINKED ISSUE #26333: OpenTelemetry with reactive rest client not working properly
### Describe the bug

Combining OpenTelemetry with the reactive rest client doesn't seem to be working correctly. Traces and spans are missing.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

I saw this as I was integrating OpenTelemetry into the super heroes `rest-fights` service when it makes downstream calls to both the `rest-heroes` & `rest-villains` services (see https://github.com/quarkusio/quarkus-super-heroes/issues/2#issuecomment-1164985577). I dumbed it down into a more simple & reproducible example:

Reproducer: https://github.com/edeandrea/otel-uni-combine

1. Clone the project & `cd` into it
2. In a terminal, run `docker compose up` to start jaeger & otel-collector
3. In another terminal, run `quarkus dev` to start the app
4. Using some rest client (`curl`, `httpie`, postman, etc), invoke a `GET` to http://localhost:8080/fruits
5. Open the Jaeger UI at http://localhost:16686
6. In the Jaeger **Search** field, select the `otel-uni-combine` service
7. In the Jaeger **Operation**, select the `/fruits` operation
8. Click on the trace
   trace

9. Notice that the trace details only include the call to `/apple` and not to `/pear`. The trace is lost somewhere.
   trace-details

### Output of `uname -a` or `ver`

Darwin edeandrea-m1pro 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000 arm64

### Output of `java -version`

```
openjdk version "17.0.3" 2022-04-19
OpenJDK Runtime Environment Temurin-17.0.3+7 (build 17.0.3+7)
OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode)
```

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/17.0.3-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.4", arch: "aarch64", family: "mac"
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #25336
- Fixes #26333

### CODE REVIEW

**Comment by gsmet** [comment]
@brunobat could you have a look at this one in priority? I would like to include it in the upcoming 2.10.1.Final.

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java [hunk 1]
```diff
@@ -56,7 +56,11 @@ public ClientRequestContextImpl(RestClientRequestContext restClientRequestContex
 
         // Capture or create a duplicated context, and store it.
         Context current = client.vertx.getOrCreateContext();
-        this.context = VertxContext.getOrCreateDuplicatedContext(current);
+        if (VertxContext.isDuplicatedContext(current)) {
+            this.context = VertxContext.createNewDuplicatedContext();
```
Comment (cescoffier) [inline independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java:60]: The problem here is that we check that `current` is a duplicated context, but create a new one from Vertx.currentContext(), which may not be `current`

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java (hunk 1 above)
Comment (radcortez) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java:60]: Yes, that is why we need the API that creates a new duplicated context from a specified context. Right now that is the only one we have.

Comment (cescoffier) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java:60]: or merge the PR I just opened on smallrye commons :-D

Comment (radcortez) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java:60]: Thanks :)

**Review by cescoffier — APPROVED** [review]
I just added a comment, the rest looks good.

File: independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java [hunk 2]
```diff
@@ -56,7 +56,7 @@ public ClientRequestContextImpl(RestClientRequestContext restClientRequestContex
 
         // Capture or create a duplicated context, and store it.
```
[the remark this answers is not in the document]
Comment (radcortez) [reply independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/ClientRequestContextImpl.java]: Updated.

**Review by brunobat — APPROVED** [review]
Looks ok to me. Thanks @radcortez

**Pull request merged**